generation/maisi/scripts/diff_model_infer.py

```diff
@@ -285,6 +286,9 @@ def diff_model_infer(env_config_path: str, model_config_path: str, model_def_pat
     )
     save_image(data, output_size, out_spacing, output_path, logger)
 
+    if dist.is_initialized():
```

Comment:
It is necessary to add device information in `output_path` avoiding overriding when multi-GPU inference.

Reply:
Added the local rank info in the output path, please help take a look whether it make senses to you, thanks.